From 33c2575ebf9c8022521d36f51b9b31cd41f7f74f Mon Sep 17 00:00:00 2001
From: Vishvananda Ishaya <vishvananda@gmail.com>
Date: Thu, 21 Jun 2012 13:25:57 -0700
Subject: [PATCH] Call libvirt_volume_driver with right mountpoint

 * fixes bug 1013782
 * fixes tests for live migration
 * removed test which doesn't apply

(cherry picked from commit 96c86336c69b9d456e43234e3fe315bd3b101045)

Change-Id: I8f95c6baa7aad878af19d5d8b8b34531a4a43885
---
 nova/tests/test_libvirt.py      |    3 ++-
 nova/virt/libvirt/connection.py |    8 ++++----
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/nova/tests/test_libvirt.py b/nova/tests/test_libvirt.py
index af30051..ec8ae9a 100644
--- a/nova/tests/test_libvirt.py
+++ b/nova/tests/test_libvirt.py
@@ -1194,7 +1194,8 @@ class LibvirtConnTestCase(test.TestCase):
         self.mox.StubOutWithMock(conn, "volume_driver_method")
         for v in vol['block_device_mapping']:
             conn.volume_driver_method('connect_volume',
-                                     v['connection_info'], v['mount_device'])
+                                     v['connection_info'],
+                                     v['mount_device'].rpartition("/")[2])
 
         # Starting test
         self.mox.ReplayAll()
diff --git a/nova/virt/libvirt/connection.py b/nova/virt/libvirt/connection.py
index e0943ba..5ec029a 100644
--- a/nova/virt/libvirt/connection.py
+++ b/nova/virt/libvirt/connection.py
@@ -473,10 +473,10 @@ class LibvirtConnection(driver.ComputeDriver):
             block_device_info)
         for vol in block_device_mapping:
             connection_info = vol['connection_info']
-            mountpoint = vol['mount_device']
+            mount_device = vol['mount_device'].rpartition("/")[2]
             self.volume_driver_method('disconnect_volume',
                                       connection_info,
-                                      mountpoint)
+                                      mount_device)
         if cleanup:
             self._cleanup(instance)
 
@@ -2126,10 +2126,10 @@ class LibvirtConnection(driver.ComputeDriver):
             block_device_info)
         for vol in block_device_mapping:
             connection_info = vol['connection_info']
-            mountpoint = vol['mount_device']
+            mount_device = vol['mount_device'].rpartition("/")[2]
             self.volume_driver_method('connect_volume',
                                       connection_info,
-                                      mountpoint)
+                                      mount_device)
 
     def pre_block_migration(self, ctxt, instance_ref, disk_info_json):
         """Preparation block migration.
